src/hotspot/share/adlc/adlArena.cpp

```diff
@@ -129,28 +130,41 @@ void* AdlArena::grow( size_t x ) {
 //------------------------------calloc-----------------------------------------
 // Allocate zeroed storage in AdlArena
 void *AdlArena::Acalloc( size_t items, size_t x ) {
+  assert(items * x <= (size_t)1 << 31, "unreasonable arena allocation size");
+  assert(items <= SIZE_MAX / x, "overflow");
   size_t z = items*x;   // Total size needed
   void *ptr = Amalloc(z);       // Get space
   memset( ptr, 0, z );          // Zap space
   return ptr;                   // Return space
 }
 
 //------------------------------realloc----------------------------------------
+// Offsets a pointer with saturation to prevent overflow.
+uintptr_t saturated_pointer_add(uintptr_t ptr, size_t size) {
+  if (ptr > UINTPTR_MAX - size) {
+    return UINTPTR_MAX;
+  }
+  return ptr + size;
+}
+
 // Reallocate storage in AdlArena.
 void *AdlArena::Arealloc( void *old_ptr, size_t old_size, size_t new_size ) {
   char *c_old = (char*)old_ptr; // Handy name
   // Stupid fast special case
-  if( new_size <= old_size ) {  // Shrink in-place
-    if( c_old+old_size == _hwm) // Attempt to free the excess bytes
-      _hwm = c_old+new_size;    // Adjust hwm
+  if (new_size <= old_size) {     // Shrink in-place
+    if (c_old + old_size == _hwm) // Attempt to free the excess bytes
+      _hwm = c_old + new_size;    // Adjust hwm
```

Comment:
This `if` is missing braces around the consequent.  If fixing the whitespace, the missing braces should
also be added.